gtsam/navigation/tests/testNavStateImuEKF.cpp

```diff
@@ -86,25 +87,44 @@ TEST(NavStateImuEKF, DynamicsJacobian) {
 /* ************************************************************************* */
 TEST(NavStateImuEKF, PredictMatchesExplicitIntegration) {
   using namespace nontrivial_navstate_example;
-  double dt = 0.02;
-
-  // Explicit integration as per Derek's code (from raw inputs)
-  Rot3 dR = Rot3::Expmap(omega_b * dt);
-  Rot3 R_new = R0.compose(dR);  // R_new = R * dR
-  Vector3 a_world = R0 * f_b + params->n_gravity;
-  Vector3 v_new = v0 + a_world * dt;
-  Point3 p_new = p0 + v0 * dt + a_world * (0.5 * dt * dt);
-  NavState X_explicit(R_new, p_new, v_new);
+  double dt = 10;
+
+  // Explicit integration from paper
+  const Vector3 phi_b = omega_b * dt;
+  const so3::DexpFunctor local(phi_b);
+  const Matrix3 J_l = local.Jacobian().left();
+  const Matrix3 N_l = local.Gamma().left();
+
+  const NavState U{Rot3::Expmap(phi_b),  // R_new
+                   N_l * f_b * dt * dt,  // p_new
+                   J_l * f_b * dt};      // v_new
+
+  // Check against static IMU function:
+  const NavState U_static = NavStateImuEKF::IMU(omega_b, f_b, dt);
+  EXPECT(assert_equal(U, U_static, 1e-9));
+
+  // Check against Gal3::Expmap
+  Vector10 xi;
+  xi << omega_b, f_b, Vector3::Zero(), 1.0;
+  const Gal3 T = Gal3::Expmap(xi * dt);
+  const NavState U_gal3{T.rotation(), T.position(), T.velocity()};
+  EXPECT(assert_equal(U_gal3, U, 1e-9));
```

Comment:
Would it make more sense to be comparing `U_gal3` to `U_static`?  Not that it matters in anyway.

Reply:
Transitivity :-)